### PR TITLE
[platform] Added new test case "test_show_platform_pcieinfo" which will verify command: "show platform pcieinfo"

### PR DESCRIPTION
**Pull request by ppikh**
### Description of PR
Added new test case "test_show_platform_pcieinfo" which will verify command: "show platform pcieinfo"

Command "show platform pcieinfo" was not covered by CLI tests, added test with basic coverage for this command.

Summary: Added new test case "test_show_platform_pcieinfo"
Fixes # (issue)

### Type of change
- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [x] 202205

#### What is the motivation for this PR?
Command "show platform pcieinfo" was not covered by CLI tests, added test with basic coverage for this command.

#### How did you do it?
Added new test case "test_show_platform_pcieinfo"

#### How did you verify/test it?
Executed all tests from file: platform_tests/cli/test_show_platform.py

### CODE REVIEW

**Review by AntonHryshchuk — DISMISSED** [review]

**Comment by liat-grozovik** [comment]
@prgeor , @keboliu for awarness

**Review by liat-grozovik — APPROVED** [review]

File: tests/platform_tests/cli/test_show_platform.py [hunk 1]
```diff
@@ -421,3 +421,17 @@ def test_show_platform_firmware_status(duthosts, enum_rand_one_per_hwsku_hostnam
...
+    """
+    @summary: Verify output of `show platform pcieinfo`
+    """
+    duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+    cmd = "show platform pcieinfo"
+
+    logging.info("Verifying output of '{}' on '{}' ...".format(cmd, duthost.hostname))
+    pcieinfo_output = duthost.command(cmd)["stdout"]
+
+    line_regexp = '=+Display PCIe Device=+'
```
Comment (prgeor) [inline tests/platform_tests/cli/test_show_platform.py:436]: @ppikh How do we know if the PCIe device is missing via this test?

**Comment by ppikh** [comment]
Hi,

We do not validate pci devices here. We just check that command executed successfully and did not return error or traceback. In future we can add validation for PCI devices - but it should be per vendor and per device. For now only basic coverage for CLI command.

Надіслано з iPhone

7 бер. 2023 р. о 03:03 Prince George ***@***.***> пише:

﻿

@prgeor commented on this pull request.

________________________________

In tests/platform_tests/cli/test_show_platform.py<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fsonic-net%2Fsonic-mgmt%2Fpull%2F7642%23discussion_r1127228396&data=05%7C01%7Cpetrop%40nvidia.com%7Ce6b722ffd7c7432e1d6408db1ea7c485%7C43083d15727340c1b7db39efd9ccc17a%7C0%7C0%7C638137478129826855%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=k8rU82Ji%2FyNm1r6WuFXy%2F5X2DP%2FO8s51FcOyUpNawkk%3D&reserved=0>:

> @@ -421,3 +421,17 @@ def test_show_platform_firmware_status(duthosts, enum_rand_one_per_hwsku_hostnam
     verify_show_platform_firmware_status_output(firmware_output_lines, duthost.hostname)

     # TODO: Test values against platform-specific expected data
+
+
+def test_show_platform_pcieinfo(duthosts, enum_rand_one_per_hwsku_hostname):
+    """
+    @summary: Verify output of `show platform pcieinfo`
+    """
+    duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+    cmd = "show platform pcieinfo"
+
+    logging.info("Verifying output of '{}' on '{}' ...".format(cmd, duthost.hostname))
+    pcieinfo_output = duthost.command(cmd)["stdout"]
+
+    line_regexp = '=+Display PCIe Device=+'


@ppikh<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fppikh&data=05%7C01%7Cpetrop%40nvidia.com%7Ce6b722ffd7c7432e1d6408db1ea7c485%7C43083d15727340c1b7db39efd9ccc17a%7C0%7C0%7C638137478129826855%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=RyMuU0nzppjLKUMCI6%2FARAYc3y%2BFP%2FP9dWIGlj6WDpI%3D&reserved=0> How do we know if the PCIe device is missing via this test?

**Comment by prgeor** [comment]
> Hi, We do not validate pci devices here. We just check that command executed successfully and did not return error or traceback. In future we can add validation for PCI devices - but it should be per vendor and per device. For now only basic coverage for CLI command.

@ppikh This test is incomplete without checking for the PCIe endpoint on the platform. Please update the test to check with `-c` option as shown below:

We don't need a separate PR just to add the check to see if all the endpoints have "Passed" 

image

File: tests/platform_tests/cli/test_show_platform.py (hunk 1 above)
Comment (ppikh) [reply tests/platform_tests/cli/test_show_platform.py]: I've updated test case

**Comment by ppikh** [comment]
Hi @prgeor  I've updated test case to check "Passed". Please check and merge code. Thanks

**Review by prgeor — APPROVED** [review]

**Pull request merged**